pkg/volume/util.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+func GetAccessModesAsString(modes []api.AccessModeType) string {
+	modesAsString := ""
+
+	if contains(modes, api.ReadWriteOnce) {
+		appendAccessMode(modesAsString, "RWO")
+	}
+	if contains(modes, api.ReadOnlyMany) {
+		appendAccessMode(modesAsString, "ROX")
+	}
+	if contains(modes, api.ReadWriteMany) {
+		appendAccessMode(modesAsString, "RWX")
+	}
+
+	return modesAsString
+}
+
+func appendAccessMode(modes, mode string) string {
+	if modes != "" {
+		modes += ","
+	}
+	modes += mode
+	return modes
+}
+
+func contains(modes []api.AccessModeType, mode api.AccessModeType) bool {
+	for _, m := range modes {
+		if m == mode {
+			return true
+		}
+	}
+	return false
+}
+
+// TODO refactor this to get this info from the VolumePluginManager
+func GetAccessModeType(source api.PersistentVolumeSource) []api.AccessModeType {
```

Comment:
@thockin what do you think about saving this on the PV when it gets created?  The volume instance takes its access modes at birth from the volume plugin.  Then it's part of the status coming over to the client.